src/main/js/controls/Line/helpers/helpers.js

```diff
@@ -250,6 +250,79 @@ const draw = (scale, config, canvasSVG, dataTarget) => {
             .remove();
     }
 };
+
+/**
+ * Extracts regions inside value and prepare a list of value region
+ *
+ * @param {object} dataTarget - Data points object
+ * @param {Function} getXDataValues - checks if the value is valid and etract value for x-axis
+ *
+ * @returns {Array} value region subset
+ */
+const getValueRegionSubset = (dataTarget, getXDataValues) => {
+    const valueRegionSubset = [];
+    const valueRegions = [];
+    let valueRegion = {
+        color: undefined,
+        values: []
+    };
+    let previousColor;
+    dataTarget.values.forEach((value) => {
+        if (
+            !utils.isEmpty(value.region) &&
+            !utils.isEmpty(value.region.start) &&
+            !utils.isEmpty(value.region.end)
+        ) {
+            // If the color is different, then move it to new region set
+            if (previousColor !== value.region.color) {
+                if (valueRegion.values.length > 0) {
+                    valueRegionSubset.push(valueRegion);
+                }
+                previousColor = value.region.color;
```

Comment:
is this line necessary?  It should be taken care of below on line 287

Reply:
I agree. It is not required. I have removed this line.